### PR TITLE
use one consumer for metric gathering

### PR DESCRIPTION
**Pull request by emord**
This should make the metrics stop failing. 

Currently it uses the pillows consumer, meaning that it creates a new connection for every pillow. 

This changes it to pass in a consumer so we're only using one consumer for all metrics.

### CODE REVIEW

File: corehq/ex-submodules/pillowtop/utils.py [hunk 1]
```diff
@@ -145,15 +146,29 @@ def safe_force_seq_int(seq, default=None):
...
 
-def get_pillow_json(pillow_config):
+def get_pillow_json(pillow_config, consumer=None):
     assert isinstance(pillow_config, PillowConfig)
 
+    def _is_kafka(checkpoint):
+        return checkpoint.sequence_format == 'json'
+
     pillow = pillow_config.get_instance()
+    if consumer is None:
```
Comment (jmtroth0) [inline corehq/ex-submodules/pillowtop/utils.py]: Probably could also put this into `_is_kafka`, but I'm not sure if it's even necessary. If it's not passed in, are you ever going to need it?

**Review by jmtroth0 — APPROVED** [review]
Just a small thought

File: corehq/ex-submodules/pillowtop/utils.py [hunk 2]
```diff
@@ -172,11 +187,19 @@ def get_pillow_json(pillow_config):
         seconds_since_last = 0
         prettified_time_since_last = ''
         hours_since_last = None
-    offsets = pillow.get_change_feed().get_latest_offsets_json()
+
+    if _is_kafka(checkpoint):
+        # breaking composition boundaries so that we can use one Kafka connection
+        offsets = {
+            "{},{}".format(tp.topic, tp.partition): offset
+            for tp, offset in consumer.end_offsets(checkpoint.wrapped_sequence.keys()).items()
```
Comment (nickpell) [inline corehq/ex-submodules/pillowtop/utils.py]: @emord modernize wants to wrap ```checkpoint.wrapped_sequence.keys()``` with ```list()```.  Is ```checkpoint.wrapped_sequence``` a dict?  If so I'll go with ```six.iterkeys()```.

Comment (emord) [reply corehq/ex-submodules/pillowtop/utils.py]: it appears that it is a dict. I think it should be fine to go with `list` to be safe though

**Pull request merged**